fhir-swagger-generator/src/main/java/com/ibm/fhir/openapi/generator/FHIROpenApiGenerator.java

```diff
@@ -63,112 +116,234 @@
+            throw new Error(e);
         }
+    }
 
-        JsonObjectBuilder tag = factory.createObjectBuilder();
-        tag.add("name", "Other");
-        tags.add(tag);
+    private static void generateBatchTransactionOpenApi() throws Exception, ClassNotFoundException, Error {
+        JsonObjectBuilder swagger = factory.createObjectBuilder();
+        swagger.add("swagger", "2.0");
+        
+        JsonObjectBuilder info = factory.createObjectBuilder();
+        info.add("title", "Batch/Transaction");
+        info.add("description", "The batch and transaction interactions submit a set of actions to perform on a server in a single HTTP request/response.");
+        info.add("version", "4.0.0");
+        swagger.add("info", info);
 
-        // FHIR metadata operation
-        JsonObjectBuilder path = factory.createObjectBuilder();
-        generateMetadataPathItem(path);
-        paths.add("/metadata", path);
+        swagger.add("basePath", "/fhir-server/api/v4");
 
-        // FHIR batch operation
-        path = factory.createObjectBuilder();
+        JsonObjectBuilder paths = factory.createObjectBuilder();
+        JsonObjectBuilder definitions = factory.createObjectBuilder();
+        
+        // FHIR batch/transaction interaction
+        JsonObjectBuilder path = factory.createObjectBuilder();
         generateBatchPathItem(path);
         paths.add("/", path);
-
-        // TODO: FHIR transaction operation
-
-        swagger.add("tags", tags);
         swagger.add("paths", paths);
-
-        JsonObjectBuilder components = factory.createObjectBuilder();
-        JsonObjectBuilder parameters = factory.createObjectBuilder();
-        generateParameters(parameters, filter);
-        JsonObject parametersObject = parameters.build();
-        if (!parametersObject.isEmpty()) {
-            components.add("parameters", parametersObject);
+        
+        
+        generateDefinition(CapabilityStatement.class, definitions);
+        
+        // generate definition for all the defined Types.
+        for (String typeClassName : getAllTypesList()) {
+            Class<?> typeModelClass = Class.forName(TYPEPACKAGENAME + "." + typeClassName);
+            // System.out.println("Type:  " + className);
+            generateDefinition(typeModelClass, definitions);
         }
-        components.add("requestBodies", requestBodies);
-        components.add("schemas", definitions);
-
-        swagger.add("components", components);
-
+        
+        // generate definition for all inner classes inside the top level resources.
+        for (String innerClassName : getAllResourceInnerClasses()) {
+            Class<?> parentClass = Class.forName(RESOURCEPACKAGENAME + "." + innerClassName.split("\\$")[0]);
+            if (CapabilityStatement.class.equals(parentClass)) {
+                Class<?> innerModelClass = Class.forName(RESOURCEPACKAGENAME + "." + innerClassName);
+                // System.out.println("Resource:  " + className);
```

Comment:
can we remove the commented out System.out.println? there are a few in this file.

Reply:
done